src/parthenon_manager.cpp

```diff
@@ -242,10 +242,12 @@ void ParthenonManager::RestartPackages(Mesh &rm, RestartReader &resfile) {
     size_t index = 0;
     for (auto &pmb : rm.block_list) {
       bool found = false;
-      auto cX = MeshBlockDataIterator<Real>(
-          pmb->meshblock_data.Get(),
-          {parthenon::Metadata::Independent, parthenon::Metadata::Restart}, true);
-      for (auto &v : cX.vars) {
+      const auto this_indep_restart_vars =
+          pmb->meshblock_data.Get()
+              ->GetVariablesByFlag(
+                  {parthenon::Metadata::Independent, parthenon::Metadata::Restart}, false)
```

Comment:
Comment for a future potential improvement beyond this PR: Add an enum class or similar for `Match::Any` and `Match::All` instead of the raw `bool` directly in the interface.